helix-term/src/keymap.rs

```diff
@@ -179,21 +185,22 @@ impl DerefMut for KeyTrieNode {
 #[serde(untagged)]
 pub enum KeyTrie {
     Leaf(Command),
+    Sequence(Vec<Command>),
```

Comment:
Renaming it to something like `SequencedLeaf` would remind that it is simply another type of leaf.

Reply:
Do we need a separate variant here?  We could just have `Leaf` itself always store a `Vec<Command>`, and single commands just result in single-element `Vec`s.  And that way we don't have as many separate code paths to deal with.

Reply:
@cessen Ah, that could work. What have to be done to make that work with serde?

Reply:
In the vast majority of cases (and in the default keymap), there will be a single command and not a sequence. Hence if we change `Leaf` to be `Vec<Command>` we will be allocating `Vec`s unnecessarily all over the place (I'm not sure if the cost is negligible, but it seems like a cost nonetheless).

Reply:
True

Reply:
I don't feel strongly about, especially since it's a stop-gap feature anyway.  So feel free to ignore me. :-)

But I strongly suspect the extra `Vec`s won't be a problem in this case.  Keymaps aren't that large (a few hundred items at most), so I'd be shocked if the wasted space was significant.  And none of these items are used in tight loops where the pointer indirection would cause performance issues.

Reply:
I think I agree with @sudormrfbin since, as they said, most key mappings will be singular commands and not sequences, so it's probably not really a great idea to allocate a Vec for all of them.

Reply:
Okay, sounds good!  I do want to clarify, however, that I'm not coming from a "optimization doesn't matter" viewpoint.  I care a lot about performance and memory usage (see e.g. Ropey's design and implementation).  It's just that in this particular case I don't think it will make any meaningful difference.

But, again, I don't feel strongly about it, so I'm happy leaving it as-is as well. :-)